nibabel/freesurfer/io.py

```diff
@@ -73,6 +73,27 @@ def _read_volume_info(fobj):
     return volume_info
 
 
+def _pack_rgba(rgba):
+    """Used by :meth:`read_annot` and :meth:`write_annot` to pack an RGBA
+    sequence into a single integer.
+
+    Parameters
+    ----------
+
+    rgba : ndarray, shape (n, 4)
+        RGBA colours
+
+    Returns
+    -------
+
+    out : ndarray, shape (n, )
+        Annotation values for each colour.
+    """
+    # ctab :: n x 4
+    bitshifts = 2 ** np.array([[0], [8], [16], [24]], dtype=rgba.dtype)
+    return rgba.dot(bitshifts).squeeze()
```

Comment:
I think if you flatten this in the one case, the two cases where you have to reshape will come out a lot cleaner.

Reply:
Sorry, I think that was badly worded, and badly counted, as well. If you don't flatten here, then you can cover the two cases like so:

```Python
# Setting vector - using [4] instead of [4] maintains the shape
ctab[:, [4]] = _pack_rgba(ctab[:, :4])

# Stacking matrix and column vector
ctab = np.hstack((ctab[:, :4], _pack_rgba(ctab[:, :4])))

Reply:
Cool, makes the code a little cleaner...